client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransportFactory.java

```diff
@@ -451,35 +468,104 @@ public SSLContext getSSLContext() {
    @Override
    public void reset(byte[] cacheName) {
       updateServers(initialServers, cacheName, true);
-      topologyInfo.setTopologyId(HotRodConstants.DEFAULT_CACHE_TOPOLOGY);
+      topologyInfo.setTopologyId(cacheName, HotRodConstants.DEFAULT_CACHE_TOPOLOGY);
    }
 
    @Override
-   public boolean trySwitchCluster(byte[] cacheName) {
-      if (clusters.isEmpty()) {
-         log.debugf("No alternative clusters configured, so can't switch cluster");
-         return false;
-      }
+   public AtomicInteger createTopologyId(byte[] cacheName) {
+      return topologyInfo.createTopologyId(cacheName, -1);
+   }
 
-      if (clustersViewed >= clusters.size()) {
-         log.debugf("All cluster addresses viewed (number=%d) and none worked: %s", clustersViewed, clusters);
-         return false;
-      }
+   @Override
+   public int getTopologyId(byte[] cacheName) {
+      return topologyInfo.getTopologyId(cacheName);
+   }
```

Comment:
this method and ```createTopologyId``` interact with ```topologyInfo``` without a lock, there might be a race condition here